app/assets/javascripts/common/password_validation.js

```diff
@@ -1,35 +1,26 @@
 PasswordValidation = function() {
   const DebounceTimeout = 500;
+  const Validations = ["too_short", "needs_number", "needs_lower", "needs_upper"];
 
-  this.initialize = function() {
+  this.initialize = () => {
     this.timer = null;
-    this.response = null;
     this.passwordInput = $("#password");
-    this.passwordInput.on("input", this.handlePasswordInput);
+    this.passwordInput.on("input", this.debounce);
   }
 
-  this.handlePasswordInput = () => {
-    console.log("CHANGE", this.passwordInput.val());
-    this.setTimer();
-  }
-
-  this.setTimer = function() {
-    console.log("STARTING TIMER")
-    this.cancelTimer();
-    this.timer = setTimeout(this.validatePassword, DebounceTimeout);
-  }
+  this.debounce = () => {
+    const later = () => {
+      clearTimeout(this.timer);
+      this.validatePassword();
+    };
 
-  this.cancelTimer = function() {
-    console.log("CANCELING TIMER")
-    if (!this.timer) return;
     clearTimeout(this.timer);
-    this.timer = null;
+    this.timer = setTimeout(later, DebounceTimeout);
   }
 
   this.validatePassword = () => {
-    console.log("MAKING REQUEST")
-    const token = $("meta[name=csrf-token]").attr("content")
-    const url = "http://localhost:3000/email_authentications/validate"
+    const token = $("meta[name=csrf-token]").attr("content");
+    const url = "http://localhost:3000/email_authentications/validate";
```

Comment:
Should this be `localhost`?

Reply:
OMG. I can't believe I forgot to change that. Thank you for pointing this out. I'll fix it.

Reply:
Don't sweat it, that's what code review is for 😄